### PR TITLE
Add aerospike plugin support

### PR DESCRIPTION
**Pull request by oldmantaiter**
- Does not use the aerospike client, but sends the stats command
  using the aerospike required format
- Queries available namespaces and gets stats for all of them

### CODE REVIEW

File: plugins/aerospike/aerospike.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+	"encoding/binary"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+import (
+	"github.com/influxdb/telegraf/plugins"
```
Comment (sparrc) [inline plugins/aerospike/aerospike.go]: why separated?

**Comment by sparrc** [comment]
Thanks @oldmantaiter!, just a couple things:

[] Write a README, see https://github.com/influxdb/telegraf/blob/master/plugins/system/CPU_README.md for an example
[] More unit tests. You have quite a lot of easily testable functions in your plugin. There shouldn't only be an integration test in this case.

File: plugins/aerospike/aerospike.go (hunk 1 above)
Comment (oldmantaiter) [inline plugins/aerospike/aerospike.go]: Just the way I do it, anything that needs a go get is in the second import statement. Looking around it seems like that clashes with the current style, so I will change it to conform with that.

**Comment by oldmantaiter** [comment]
@sparrc Definitely forgot about the README, will add one. I will work on adding more unittests (they are not my forté but I will give it a shot).

**Comment by oldmantaiter** [comment]
@sparrc Added README, and a few more tests. I believe I have covered most of the functions that are in there besides the ones that call the server directly/indirectly. Let me know what you think.

**Comment by sparrc** [comment]
awesome, thanks @oldmantaiter, will get this merged soon

**Comment by oldmantaiter** [comment]
@sparrc Great! I do know that there are a lot of metrics that would get pushed because of this, I was thinking of adding some sort of globbing filter to match keys from the aerospike stats output.

Also wanted to know if instead of doing this via the stdlibs if I should just include the aerospike client library as a dependency; I was not sure if this is something that is currently allowed or if it should be avoided as much as possible.

**Pull request closed without merging**